applications/nrf_desktop/src/modules/hid_state.c

```diff
@@ -1381,9 +1383,7 @@ static bool handle_ble_peer_event(const struct ble_peer_event *event)
 static bool handle_usb_hid_event(const struct usb_hid_event *event)
 {
 	if (event->enabled) {
-		if (!get_subscriber_by_type(true)) {
-			connect_subscriber(event->id, true, 1);
-		}
+		connect_subscriber(event->id, true, 1);
```

Comment:
the check for subscriber presence was for a reason - DESK-478
in general due to problems of usb driver, state machine could duplicate the event about usb being powered

are you 100% sure this will not occur?

Reply:
I think this should no longer occur. In usb_state we send usb_hid_event only if the enabled state has changed.

Reply:
we always did; I don't recall how the bug occur but it did occur somehow
it is possible that the situation is different - if we keep it we must keep an eye on this part

Reply:
I added assert when there is no space for adding a subscriber to make sure we will notice the problem when it occurs.

Reply:
We track the subscriber enabled state in usb_state, so we should not observe multiple subscriptions for USB HID device.